src/framework-configuration.js

```diff
@@ -20,18 +20,30 @@ function runTasks(config, tasks) {
   return next();
 }
 
-function loadPlugin(config, loader, info) {
+interface FrameworkPluginInfo {
+  moduleId?: string;
+  resourcesRelativeTo?: string[];
+  configure?: (config: FrameworkConfiguration, pluginConfig?: any) => any;
+  config?: any;
+}
+
+function loadPlugin(config: FrameworkConfiguration, loader: Loader, info: FrameworkPluginInfo) {
   logger.debug(`Loading plugin ${info.moduleId}.`);
-  config.resourcesRelativeTo = info.resourcesRelativeTo;
+  if (typeof info.moduleId === 'string') {
+    config.resourcesRelativeTo = info.resourcesRelativeTo;
 
-  let id = info.moduleId; // General plugins installed/configured by the end user.
+    let id = info.moduleId; // General plugins installed/configured by the end user.
 
-  if (info.resourcesRelativeTo.length > 1 ) { // In case of bootstrapper installed plugins like `aurelia-templating-resources` or `aurelia-history-browser`.
-    return loader.normalize(info.moduleId, info.resourcesRelativeTo[1])
-      .then(normalizedId => _loadPlugin(normalizedId));
-  }
+    if (info.resourcesRelativeTo.length > 1 ) { // In case of bootstrapper installed plugins like `aurelia-templating-resources` or `aurelia-history-browser`.
+      return loader.normalize(info.moduleId, info.resourcesRelativeTo[1])
+        .then(normalizedId => _loadPlugin(normalizedId));
+    }
 
-  return _loadPlugin(id);
+    return _loadPlugin(id);
+  } else if (typeof info.configure === 'function') {
+    // use info.config || {} to keep behavior consistent with loading from string
+    return Promise.resolve(info.configure.call(null, config, info.config || {}));
+  }
```

Comment:
else? There's no return, nothing, if `configure` was neither a `string` nor a `function`. 

That's a user error, we should at least log something to help users find and fix their mistake. Otherwise it'll be confusing why features silently don't work.

Reply:
That's true, it isn't good to keep going here with an invalid configuration, no matter how it got in there.